drm-ffi/src/syncobj.rs

```diff
@@ -39,54 +42,91 @@ pub fn destroy(fd: BorrowedFd<'_>, handle: u32) -> io::Result<drm_syncobj_destro
     Ok(args)
 }
 
-/// Exports a syncobj as an inter-process file descriptor or as a poll()-able sync file.
-pub fn handle_to_fd(
+/// Exports a syncobj as an inter-process file descriptor.
+pub fn handle_to_fd(fd: BorrowedFd<'_>, handle: u32) -> io::Result<OwnedFd> {
+    let mut args = drm_syncobj_handle {
+        handle,
+        flags: 0,
+        fd: 0,
+        pad: 0,
+        point: 0,
+    };
+
+    unsafe {
+        ioctl::syncobj::handle_to_fd(fd, &mut args)?;
+    }
+
+    Ok(unsafe { OwnedFd::from_raw_fd(args.fd) })
+}
+
+/// Exports a syncobj as a poll-able sync file optionally for a specified timeline point.
+pub fn handle_to_sync_file(
     fd: BorrowedFd<'_>,
     handle: u32,
-    export_sync_file: bool,
-) -> io::Result<drm_syncobj_handle> {
+    timeline_point: Option<u64>,
+) -> io::Result<OwnedFd> {
     let mut args = drm_syncobj_handle {
         handle,
-        flags: if export_sync_file {
-            DRM_SYNCOBJ_HANDLE_TO_FD_FLAGS_EXPORT_SYNC_FILE
-        } else {
-            0
-        },
+        flags: DRM_SYNCOBJ_HANDLE_TO_FD_FLAGS_EXPORT_SYNC_FILE
+            | if timeline_point.is_some() {
+                DRM_SYNCOBJ_HANDLE_TO_FD_FLAGS_TIMELINE
+            } else {
+                0
+            },
         fd: 0,
         pad: 0,
-        point: 0, // TODO: Add support for TIMELINE sync files
+        point: timeline_point.unwrap_or(0),
     };
 
     unsafe {
         ioctl::syncobj::handle_to_fd(fd, &mut args)?;
     }
 
-    Ok(args)
+    Ok(unsafe { OwnedFd::from_raw_fd(args.fd) })
 }
 
 /// Imports a file descriptor exported by [`handle_to_fd`] back into a process-local handle.
-pub fn fd_to_handle(
+pub fn fd_to_handle(fd: BorrowedFd<'_>, opaque_fd: BorrowedFd<'_>) -> io::Result<u32> {
+    let mut args = drm_syncobj_handle {
+        handle: 0,
+        flags: 0,
+        fd: opaque_fd.as_raw_fd(),
+        pad: 0,
+        point: 0,
+    };
+
+    unsafe {
+        ioctl::syncobj::fd_to_handle(fd, &mut args)?;
+    }
+
+    Ok(args.handle)
+}
+
+/// Imports a sync file exported by [`handle_to_sync_file`] into an existing process-local handle optionally for a specified timeline point.
```

Comment:
```suggestion
/// Imports a `sync_file` exported by [`handle_to_sync_file()`] into an existing process-local handle, optionally for a specified timeline point.
```

Not sure whether `sync_file` really is the right naming, otherwise maybe `Sync File` with a link to https://docs.kernel.org/driver-api/sync_file.html?

Reply:
I think that naming is good. and documentation should be easy to find, if you deal with linux drm.